src/server/survey_and_program_data.R

```diff
@@ -193,12 +194,13 @@ surveyAndProgramData <- function(input, output, state, spectrumFilesState) {
 
         newSurvey <- read.csv(inFile$datapath, check.names=FALSE)
 
-        state$wrongSurveyHeaders <<- !identical(sort(names(newSurvey)), sort(names(state$survey_data_human_readable())))
+        state$wrongSurveyHeaders <- !identical(sort(names(newSurvey)), sort(names(state$survey_data_human_readable())))
 
-        state$wrongSurveyCountry <<- !state$wrongSurveyHeaders && nrow(subset(newSurvey, gsub("\t", "", removeSpecialChars(Country)) == removeSpecialChars(spectrumFilesState$countryAndRegionName()))) < nrow(newSurvey)
+        state$wrongSurveyCountry <- !state$wrongSurveyHeaders &&
```

Comment:
This checks that newly uploaded survey data has the correct country/region in all the rows. Similarly the logic below checks it for newly uploaded programmatic data,